Added documented example for spatial selection in tripole

aidanheerdegen opened the pull request:
Added documented example for spatial selection in tripole and adding coordinates to ice data

aidanheerdegen commented:
This notebook is viewable here:

https://nbviewer.jupyter.org/github/COSIMA/cosima-recipes/blob/5cc87907ad811603f0cd5d21f8c745629f1faef4/DocumentedExamples/Spatial_selection.ipynb

navidcy commented:
@aidanheerdegen this is great and very helpful.

aidanheerdegen commented:
Thanks for the review @navidcy 

I have made the suggested changes and explained the use of `xr.ufuncs.degrees`.

The updated notebook can be viewed here:

https://nbviewer.jupyter.org/github/COSIMA/cosima-recipes/blob/69fc67b9abc96c5e814cde8214b38d77c0346c68/DocumentedExamples/Spatial_selection.ipynb

I don't really like that in this notebook I am just grabbing some random grid from an input directory that may not exist at some point in the future.

I agree that it is problematic that there are not coordinates easily available automatically. I don't know the best solution, but I'm not sure adding more coordinates to all data files is the best solution. 

A good solution is probably linked with the idea of "hashing" grids to uniquely identify them in the database

https://github.com/COSIMA/cosima-cookbook/issues/137#issuecomment-506941594

Once you can identify a grid then it should be possible to lookup and add "good" coordinates for that grid.

Perhaps combined with the idea of adopting Xgcm grid objects

https://github.com/COSIMA/cosima-cookbook/issues/170

aidanheerdegen commented:
I meant to ping @angus-g in that last comment to give him a heads-up about the grid stuff.

navidcy commented:
@aidanheerdegen this is great. I ran the notebook myself and made sure all is good.

two more minor comments:

-remove 
```python
import pandas as pd
pd.set_option('display.max_rows', 500)
```
since now are not needed. 

- change the notebook's env to the 20.04?



Regarding the. better grid implementation etc, I see your suggestions. I agree but they go beyond the scope of this PR. This notebook would be useful for now even at its current form.

aidanheerdegen commented:
Thanks for picking the spurious import @navidcy. I have fixed it and removed the spurious `numpy` import.

I have also changed the kernel to 20.04. It is a good idea to test it with the latest available package versions, and to have the latest available kernel so that it remains valid for a longer time.

I agree that the grid stuff is not solvable in this PR, but it is good to document this stuff and link to relevant issues so this can be discovered again.

Updated notebook viewable here:

https://nbviewer.jupyter.org/github/COSIMA/cosima-recipes/blob/50cf2dc438cfef6bac43d945af7ae8198f1fa6f1/DocumentedExamples/Spatial_selection.ipynb

navidcy commented:
I'm excited to merge this.

aidanheerdegen commented:
Oh damn, I meant to ask if you thought it was appropriate as a Documented Example, rather than a Contributed one?

navidcy commented:
Isn't it a documented one? The PR title says "Documented example". It's **definitely** a document example!

aidanheerdegen commented:
Yeah it is written as one, just wanted to make sure it met that criteria.
